Bottom: 6ea9a39939c26b41cca4c7351a7fe8794fdc29de
Top:    f72e404022cc3892be6a2722d337af2cab14e9fd
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-30 20:37:26 +0100

Refresh of mymem-c-don-t-use-g_list_free_

---

diff --git a/src/mymem.c b/src/mymem.c
index 77a2ce2..1feb2ef 100644
--- a/src/mymem.c
+++ b/src/mymem.c
@@ -56,6 +56,17 @@ RCSID ("$Id$");
  */
 static void DSRealloc (DynamicStringTypePtr, size_t);
 
+
+/* This API is quite new, provide a version here */
+#if !GLIB_CHECK_VERSION (2, 28, 0)
+static void
+g_list_free_full (GList *list, GDestroyNotify free_func)
+{
+  g_list_foreach (list, (GFunc) free_func, NULL);
+  g_list_free (list);
+}
+#endif
+
 /* ---------------------------------------------------------------------------
  * get next slot for a rubberband connection, allocates memory if necessary
  */
